addons/web/static/src/views/fields/dynamic_widget/dynamic_widget.js

```diff
@@ -0,0 +1,66 @@
+/** @odoo-module */
+
+import { CharField, charField } from "@web/views/fields/char/char_field";
+import { registry } from "@web/core/registry";
+
+import { DynamicModelFieldSelector }  from "./dynamic_model_field_selector";
+
+
+export class DynamicDomainWithChar extends CharField {
+    static components = {
+        ...CharField.components,
+        DynamicModelFieldSelector
+    };
+
+    static props = {
+        ...CharField.props,
+        resModel : { type: String, optional: true },
+    }
+
+    /**
+     * Update record
+     *
+     * @param {string} value
+     * @private
+     */
+    async _onRecordUpdate(value) {
+        await this.props.record.update({[this.props.name]: value});
+        // await this.props.record.save();
+        // await this.props.record.load();
+    }
+
+    //---- Getters ----
+    get getSelectorProps() {
+        return {
+            path: this.props.record.data[this.props.name],
+            resModel: this.getResModel(),
+            readonly: this.props.readonly,
+            record: this.props.record,
+            recordProps: this.props,
+            update: this._onRecordUpdate.bind(this),
+            isDebugMode: !!this.env.debug,
+        };
+    }
+
+    getResModel(props = this.props) {
+        let resModel = props.record.data[props.resModel];
+        if (!resModel) { 
+            resModel = props.record.resModel;
+        }
+        return resModel;
+    }
+}
+
+DynamicDomainWithChar.template = "web.DynamicDomainWithChar";
+
+export const dynamicDomainWithChar = {
```

Comment:
Although it looks nice, I don't really like the complexity involved by the usage of this widget, combined to the complexity of the code managing phone numbers ... you can browse all fields, go into sub-fields, select wrong fields, fields you can't access due to ACLS, it involved a lot of configuration, ...

I think it should be simpler, like what we do with SMS: phone fields are defined on models. Actually the whole number finding is already implemented at model level, what is not working in current code (see phone_validation / sms) ?

Reply:
If you really need this field component, does it need to be defined in web/? Also you should clarify the naming (DynamicWidget, DynamicDomain, dynamicDomainWithChar...).

Reply:
@aab-odoo This is an add-in web just because in fuhrer some one can use it.